fdbcli/SetClassCommand.actor.cpp

```diff
@@ -71,15 +73,26 @@ ACTOR Future<Void> printProcessClass(Reference<IDatabase> db) {
 };
 
 ACTOR Future<bool> setProcessClass(Reference<IDatabase> db, KeyRef network_address, KeyRef class_type) {
-    state Reference<ITransaction> tr = db->createTransaction();
+	state Reference<ITransaction> tr = db->createTransaction();
 	loop {
 		tr->setOption(FDBTransactionOptions::SPECIAL_KEY_SPACE_ENABLE_WRITES);
 		try {
-            tr->set(network_address.withPrefix(fdb_cli::processClassTypeSpecialKeyRange.begin), class_type);
-            wait(safeThreadFutureToFuture(tr->commit()));
-            return true;
-        } catch (Error& e) {
-			wait(safeThreadFutureToFuture(tr->onError(e)));
+			tr->set(network_address.withPrefix(fdb_cli::processClassTypeSpecialKeyRange.begin), class_type);
+			wait(safeThreadFutureToFuture(tr->commit()));
+			return true;
+		} catch (Error& e) {
+			state Error err(e);
+			if (e.code() == error_code_special_keys_api_failure) {
+				Optional<Value> errorMsg = wait(safeThreadFutureToFuture(tr->get(fdb_cli::errorMsgSpecialKey)));
+				ASSERT(errorMsg.present());
+				std::string errorMsgStr;
+				auto valueObj = readJSONStrictly(errorMsg.get().toString()).get_obj();
+				auto schema = readJSONStrictly(JSONSchemas::managementApiErrorSchema.toString()).get_obj();
+				// error message already has \n at the end
+				fprintf(stderr, "%s", valueObj["message"].get_str().c_str());
+				return false;
+			}
```

Comment:
This is really useful! I checked out this PR and tried `setclass abc storage` and it gave me the nice error message `ERROR: 'abc' is not a valid network endpoint address`, whereas before I only got `ERROR: Api call through special keys failed. For more information, call get on special key 0xff0xff/error_message to get a json string of the error message. (2117)`. It seems like printing the the error message will be repeated for a lot of different CLI commands, do you plan to create a function for this or reuse this block in each place?

Reply:
yeah, thanks for catching this.
I created a helper function to do the schema check and return the error message.